### ISSUE TITLE
Complete river layer

### ISSUE THREAD

**Issue by frogocomics** [Captain-Chaos/WorldPainter#65]
As a terraformer, the river layer would be extremely useful for realistic looking rivers. Is it possible that the river layer be completed and usable soon?

**Comment by Captain-Chaos** [comment]
I agree it would be! The trouble is that once you dive in there are a lot of questions about how it should work. I want it to create good looking rivers as automatically as possible. That entails for example the water level not following every bump in the terrain. But if it shouldn't do that, how should it determine what the water level _should_ be? Ideally, the water should only ever go down from the source, never up, but that would be extremely hard to do. Regardless of the answer it's likely to be complex and difficult to implement. Etc., etc.

However you've triggered me to put some time into this.

**Comment by Captain-Chaos** [comment]
It would help to hear your thoughts about how it would work. Say I paint a line of River across a landscape of gently undulating hills. What _exactly_ would you expect it to result in?

**Comment by frogocomics** [comment]
I want it to bull through terrain (follow the shape of the terrain). It would be up to the terraformer to determine best placement. A river should have variable depth, which can be painted. I would be nice to have a maximum depth (ie 7 blocks), and then you can paint it so 100% is the max depth.

**Comment by Captain-Chaos** [comment]
Is that all? ;-)

**Comment by Captain-Chaos** [comment]
Another option I have been thinking of is a tool instead of a layer, which carves out the river and fills it with water permanently in the editor. Or perhaps a combination of the two.

River support has been slowly bubbling away on a back burner for many years, and it is time to get it in some kind of usable shape.

**Comment by frogocomics** [comment]
A layer would be most desirable, as WorldMachine and other programs can generate river maps.
